test/test_type_promotion.py

```diff
@@ -430,6 +431,104 @@ def test_transpose(self, device):
         a = torch.tensor([[True, True], [False, True]], device=device)
         self.assertEqual(a.t() == 0, a.t() == False)  # noqa: E712
 
+    def _test_sparse_op_input_tensors(self, device, dtype, coalesced, zeros=True):
```

Comment:
(carrying over from previous discussion) if we want this testing to be comprehensive, then we should test tensors of varying sparsities. Right now every tensor that is being tested is 0% sparse.

Reply:
In addition, it would be nice to get a single helper, `make_sparse_tensor`, so that it can be reused beyond binary operations

Reply:
> every tensor that is being tested is 0% sparse

I agree more sparsity could help, but it's not 0%. The random numbers should include some 0's for most of these, and 0's are excluded from the sparse representation.

Reply:
Sorry, I meant that all of the dense (half, float, double) tensors are at 0% sparsity. It is improbable for a random dense tensor to have a value that is 0

Reply:
added a mask to ensure sparsity